### PR TITLE
SP2-83-Create-ProductionUnit-class

### PR DESCRIPTION
**Pull request by KonyakB**
created ProductionUnit class, if i remember correctly we agreed on having the electricity, gas and co2emission as not nullable fields since making the graphs of the different units is easier that way and brings no drawbacks.

### CODE REVIEW

**Comment by frske** [comment]
As already discussed on discord, I would suggest using primary constructors instead of the "classic constructors". 
- Would have increased readability, less boilerplate code to be written
- Is a more "modern" feature, which are to be preferred as outlined by the coding guidelines

Otherwise see no problems with the class(es)

**Comment by frske** [comment]
Also, maybe having a .ToString() override would also be useful

**Comment by magwrap** [comment]
I just renamed one commit

File: Heatington/AssetManager/ProductionUnit.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        }
+    }
+
+    public ProductionUnit(string name, string picturePath, double maxHeat, double productionCost, double maxElectricity,
+        double gasConsumption, double co2Emission)
+    {
+        Id = Guid.NewGuid();
+        Name = name;
+        PicturePath = picturePath;
+        OperationPoint = 1; // operation point set to 1 on default, meaning when created it is on maximum capacity
```
Comment (pkyeager) [inline Heatington/AssetManager/ProductionUnit.cs:34]: If OperationPoint is from the beginning directly 1, would this cause a problem for you @frske?

File: Heatington/AssetManager/ProductionUnit.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    public Guid Id { get; set; }
+    public string Name { get; set; }
+    public string PicturePath { get; set; }
+    private double _operationPoint;
+    public double MaxHeat { get; set; } // MW
+    public double ProductionCost { get; set; } // DKK/MWh
+    public double MaxElectricity { get; set; } // MW
+    public double GasConsumption { get; set; } // MWh(gas/oil)/MWh
+    public double Co2Emission { get; set; } // kg/MWh
+
```
Comment (pkyeager) [inline Heatington/AssetManager/ProductionUnit.cs:14]: Most of the props are not required to have a setter? If so, you could increase the level of encapsulation by only define the getter?

What do you think? @KonyakB

Comment (KonyakB) [reply Heatington/AssetManager/ProductionUnit.cs:14]: i agree, chenged them up

File: Heatington/AssetManager/ProductionUnit.cs (hunk 1 above)
Comment (frske) [reply Heatington/AssetManager/ProductionUnit.cs:34]: > If OperationPoint is from the beginning directly 1, would this cause a problem for you @frske?

not that I know of, I set it either way. 
I would put it to 0 tho, seems more natural, initialising it as inactive and then giving it's capacity.

Comment (KonyakB) [reply Heatington/AssetManager/ProductionUnit.cs]: operation point is set to 0 now on default

**Comment by KonyakB** [comment]
had to force push bc i forgot to include issue key in commit message

**Review by pkyeager — APPROVED** [review]
Nice job!

**Pull request merged**